src/Microsoft.Extensions.DependencyInjection.Abstractions/IDesignTimeServiceProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// A factory for creating the application's <see cref="IServiceProvider"/> at design time. Implement this interface
+    /// to enable design-time services that need access to the application's services. Design-time services will
+    /// automatically discover implementations of this interface.
```

Comment:
I understood you were talking about formalizing the Startup class pattern. This is much more limited. XDesProc will need another interface with the same shape to obtain the runtime IServiceProvider. Navigation uses the page type exclusively [Frame.Navigate](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.inavigate) and the system instantiates the page with that (and an optional object that is meant to be passed as-is to the page, defined on Frame). And well that part is off course shared with C++. Not sure whether that is definitively prohibitive for what I was thinking this could become.